torchtune/models/llama3/_tokenizer.py

```diff
@@ -166,11 +182,18 @@ def decode(
         Returns:
             str: The decoded string.
         """
-        return self.tt_model.decode(
-            token_ids,
+        # We will remove special tokens manually via regex on the decoded string.
+        # This is because removing all special tokens does not remove the role and
+        # whitespace added from the special tokens, i.e., the "user" and "\n\n" in
+        # "<|start_header_id|>user<|end_header_id|>\n\n"
+        decoded_string = self.tt_model.decode(
+            token_ids=token_ids,
             truncate_at_eos=truncate_at_eos,
-            skip_special_tokens=skip_special_tokens,
+            skip_special_tokens=False,
         )
+        if skip_special_tokens:
```

Comment:
So I understand that the goal here is just to fix `skip_special_tokens` for Llama3 tokenizer decode, but it seems to me like we are doing something very unexpected here with `skip_special_tokens`. (a) we have it defined on the base tokenizer and it's basically a no-op now, and (b) we are now inconsistent on whether this needs to be defined on the `ModelTokenizer` and `BaseTokenizer`. If it is a function of `tokenize_messages` on the `ModelTokenizer` moreso than the `BaseTokenizer`, maybe we should update the Protocol along with other callsites?